src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseExportService.java

```diff
@@ -132,8 +130,8 @@ public Path exportProgrammingExercise(ProgrammingExercise exercise, String pathT
 
         try {
             // Zip the student and instructor repos together.
-            // The filename of the zip is: {Exercise_Short_Name}-{Exercise_Title}.zip
-            var filename = exercise.getShortName() + "-" + exercise.getTitle() + ".zip";
+            var timestamp = ZonedDateTime.now().format(DateTimeFormatter.ofPattern("yyyyMMdd-Hmss"));
+            var filename = exercise.getCourseViaExerciseGroupOrCourseMember().getShortName() + "-" + exercise.getTitle() + "-" + timestamp + ".zip";
```

Comment:
I think this causes some of the problems described with the same archive titles. Maybe adding a prefix for the different exercise types would solve the problem?

Reply:
Wow thanks for the catch! I append the exercise id to the zip file name now in order to make each file unique